gen/function_source.c

```diff
@@ -15,13 +15,13 @@ const uint8_t qjsc_function_source[320] = {
  0x6e, 0x20, 0x66, 0x28, 0x29, 0x20, 0x7b, 0x20,
  0x72, 0x65, 0x74, 0x75, 0x72, 0x6e, 0x20, 0x34,
  0x32, 0x20, 0x7d, 0x0d, 0xc0, 0x03, 0x00, 0x00,
- 0x00, 0x00, 0x00, 0x0c, 0x20, 0xfa, 0x01, 0xa2,
+ 0x00, 0x00, 0x00, 0x0c, 0x20, 0x0a, 0x01, 0xa2,
```

Comment:
Not exactly sure what this is, but since it triggered the need for a rebuild I'll bump the BC version.

Reply:
It took me a while to figure out but it's an existing sign-extension bug that your change magically fixes.

If you apply this patch to master and run `make codegen`, you get the same output as your PR:
```diff
diff --git a/quickjs.c b/quickjs.c
index 86eee69..2edc867 100644
--- a/quickjs.c
+++ b/quickjs.c
@@ -33805,6 +33805,7 @@ static void bc_put_sleb128(BCWriterState *s, int32_t v)
 
 static void bc_set_flags(uint32_t *pflags, int *pidx, uint32_t val, int n)
 {
+    val &= (1u << n) - 1;
     *pflags = *pflags | (val << *pidx);
     *pidx += n;
 }
```
The changed field is the flags field of function objects. It's 16 bits wide, surfacing the bug, whereas other flags fields are only 8 bit wide.